components/automate-ui/src/app/components/sidebar/sidebar.component.spec.ts

```diff
@@ -34,53 +35,55 @@ describe('SidebarComponent', () => {
     featureFlags = TestBed.inject(FeatureFlagsService);
     spyOn(store, 'dispatch').and.callThrough();
   }));
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(SidebarComponent);
-    component = fixture.componentInstance;
-    // element = fixture.debugElement.nativeElement;
-
-    // enable all feature flags, if any, for testing
-    featureFlags.setFeature('servicenow_cmdb', true);
+  describe('component is created', () => {
 
-    fixture.detectChanges();
-  });
+    beforeEach(() => {
+      fixture = TestBed.createComponent(SidebarComponent);
+      component = fixture.componentInstance;
+    });
 
-  it('should be created', () => {
-    expect(component).toBeTruthy();
+    it('should be created', () => {
+      expect(component).toBeTruthy();
+    });
   });
 
+  // Only need to check components that copy the route structure defined by LayoutSidebarService.
+  // The unit tests here are a meta-check to ensure that if you make a change in one place,
+  // you must also make the same change in the copy.
   using([
-    [new SettingsLandingComponent(), Sidebar.Settings, 'Settings Landing Component'],
-    [new ComplianceLandingComponent(), Sidebar.Compliance, 'Compliance Landing Component']
+    [new SettingsLandingComponent(), Sidebar.Settings, 'servicenow_cmdb', 'Settings Landing Component'],
+    [new ComplianceLandingComponent(), Sidebar.Compliance, '', 'Compliance Landing Component']
 
-  ], function (_landingComponent: any, sidebar: Sidebar,  description: string) {
+  ], function (landingComponent: any, sidebar: Sidebar,  featureFlag: string, description: string) {
     describe(`${description} route list`, () => {
 
       beforeEach(() => {
+        if (featureFlag) {
+          featureFlags.setFeature(featureFlag, true);
+        }
+        // must be after feature flag setting!
+        layoutFacade = TestBed.inject(LayoutFacadeService);
+
+        fixture = TestBed.createComponent(SidebarComponent);
+        element = fixture.debugElement.nativeElement;
+
         layoutFacade.showSidebar(sidebar);
         fixture.detectChanges();
       });
 
-      // FIXME(tc): Sometimes missing elements. Needs investigation.
-      // it('has length consistent with sidebar', () => {
-      //   const links = element.querySelectorAll('div.nav-items chef-sidebar-entry');
-      //   expect(links.length).toBe(landingComponent.routeList.length);
-      // });
-
-      // FIXME(sr): This test randomly fails, insofar as the elements are all there,
-      //            but in the wrong order.
-      // it('has route order consistent with sidebar', () => {
-      //   const links = element.querySelectorAll('div.nav-items chef-sidebar-entry');
-      //   for (let i = 0; i < links.length; i++) {
-      //     const link: any = links[i];
-      //     expect(link.route).toBe(landingComponent.routeList[i].route);
-      //   }
-      // });
+      it('has route order consistent with sidebar', () => {
```

Comment:
The two separate tests are really different aspects of the same test, so combined them.